### PR TITLE
Fix Dnrm2 when the slice contains multiple inf.

### LINKED ISSUE #135: native/Dnrm2 returns NaN if multiple Inf values
For example,
N = 10, IncX =1, x = []float64{4.474077183601786e-75, -Inf, -Inf, 1.398043285130107e-76, 1.398043286095289e-76, 1.398043286095289e-76, 1.398043286095289e-76, 1.398043286095289e-76, 1.398043286095289e-76, 1.398043286095289e-76},

native/drnm2 returns NaN, but it should return +inf.

### PR DESCRIPTION
**Pull request by btracey**
Fixes #135

### CODE REVIEW

**Comment by btracey** [comment]
Wait. Not sufficient.

**Comment by btracey** [comment]
PTAL

At least on my machine, OpenBLAS and gonum disagree on the correct behavior with Idamax where there are NaNs present. If you'd like I can log the failure instead of causing a t.Error.

**Comment by kortschak** [comment]
LGTM with one minor optional comment and yes, I'd like a logged failure rather than an error - with a TODO to remove it when the OpenBLAS bug (yes) is fixed. Can we do that without weakening the other Idamax test cases?

File: native/level1double.go [hunk 1]
```diff
@@ -55,6 +58,9 @@ func (Implementation) Dnrm2(n int, x []float64, incX int) float64 {
 				sumSquares = sumSquares + (absxi/scale)*(absxi/scale)
 			}
 		}
+		if math.IsInf(scale, 1) {
+			return math.Inf(1)
```
Comment (kortschak) [inline native/level1double.go]: return scale rather than calling math.Inf(1)? Same for NaN and Inf elsewhere if so.

**Comment by btracey** [comment]
The fuzzer I have ( https://github.com/btracey/blasfuzz/blob/master/onevec/idamax.go ) tests differently if the slice has a NaN value or not. My plan was to do a similar test (if the slice has a NaN, log the failure).

File: testblas/common.go [hunk 2]
```diff
@@ -14,6 +14,12 @@ func dTolEqual(a, b float64) bool {
 	if math.IsNaN(a) && math.IsNaN(b) {
 		return true
 	}
+	if math.IsInf(a, 1) && math.IsInf(b, 1) {
```
Comment (kortschak) [inline testblas/common.go]: Why not just check if a == b for these two cases and return fast for all cases where a is exactly b?

**Comment by kortschak** [comment]
That approach sounds fair.

**Pull request merged**